### DISCUSSION TITLE
Duplicate extension names found

### DISCUSSION THREAD

**Discussion by kgish** [ueberdosis/tiptap#3030] [General Questions]
I have created a custom extension and it seems to be working just fine.

However, when I look in the devtools window, I see many warning being generated, namely:

```
ExtensionManager.ts:144 [tiptap warn]: Duplicate extension names found: ['paragraph', 'blockquote', 'bold',
 'bulletList', 'code', 'codeBlock', 'doc', 'dropCursor', 'gapCursor', 'hardBreak', 'heading', 'history', 
'horizontalRule', 'italic', 'listItem', 'orderedList', 'strike', 'text'].
 This can lead to issues.
```

Any idea what I might be doing wrong?

**Comment by Devrajsinh-Jhala** [comment]
Hello Sir, Today I faced the same issue, I had installed extensions for OrderedList, BulletList, Paragraph, Document, and Text so I had this type of warning. I removed the above from extensions:[] and warning disappeared and also everything is working fine right now, Please try it

**Comment by rhenck** [comment]
For me, the problem was that I used the StarterKit extension alongside other extensions that were already included in the StarterKit.

So either remove the StarterKit and add all needed extensions explicitly or check the StarterKit extensions and remove the duplicates from your extension list.

https://tiptap.dev/api/extensions/starter-kit

**Reply by Casmo** [reply]
Same here, to configure the extensions use the following (also in the docs):
```js
extensions: [
  StarterKit.configure({
    heading: {
      HTMLAttributes: {
        class: "text-xl text-gray-700",
      },
    },
  })
]
```

**Reply by quangnmwork** [reply]
Here the answer . Working for me

**Comment by ElderlyBoy** [comment]
I encountered a similar problem. I introduced `codeblocklowlight` and registered it in `extensions []`. The console outputs a warning: `Duplicate extension names found: ['codeblock '] This can lead to issues.` Later, I found that the problem lies in `Codeblocklowlight`. Its `name` attribute is not `codeblocklowlight` but `codeblock`.

**Comment by yeonjulee1005** [comment]
@ElderlyBoy I also same problem! 
`Duplicate extension names` `['codeblock]`
it seems to be compare `['codeblock-lowlight']`
is there any solutions?

**Reply by ElderlyBoy** [reply]
It seems to be a problem with the extension itself, I try to check the resource of the `CodeBlockLowlight` and then I found this:
````js
//tiptap/packages/extension-code-block-lowlight/src/code-block-lowlight.ts
export const CodeBlockLowlight = CodeBlock.extend<CodeBlockLowlightOptions>({
  addOptions() {
    return {
      ...this.parent?.(),
      lowlight: {},
      defaultLanguage: null,
    }
  },

  addProseMirrorPlugins() {
    return [
      ...this.parent?.() || [],
      LowlightPlugin({
        name: this.name,
        lowlight: this.options.lowlight,
        defaultLanguage: this.options.defaultLanguage,
      }),
    ]
  },
})
````
You can see that the name of `CodeBlockLowlight` is inherited from `CodeBlock`
So, The warning on the console seems to be OK
I tried to change the name of `CodeBlockLowlight` from `CodeBlock` to `CodeBlockLowlight`, then the console warning disappears, but I DON'T recommend.

**Reply by yeonjulee1005** [reply]
perfect! Thank you Sur!
but, it should be to change inherited name.
Should I inform the tiptap team? :)

**Reply by Spell28** [reply]
> @ElderlyBoy I also same problem! `Duplicate extension names` `['codeblock]` it seems to be compare `['codeblock-lowlight']` is there any solutions?

I solved this problem by this way:
```
StarterKit.configure({
  codeBlock: false,
}),
```

**Reply by lvbn** [reply]
This worked for me. Thanks!

**Reply by AryNoe** [reply]
Thanks a lot, this worked for me!

> > @ElderlyBoy I also same problem! `Duplicate extension names` `['codeblock]` it seems to be compare `['codeblock-lowlight']` is there any solutions?
> 
> I solved this problem by this way:
> 
> ```
> StarterKit.configure({
>   codeBlock: false,
> }),
> ```

**Comment by IliyaZafirov** [comment]
I just removed these extensions which are same as these in StarterKit.

https://tiptap.dev/docs/editor/extensions/functionality/starterkit
